pkg/controller/vsphere/reconciler.go

```diff
@@ -362,6 +338,41 @@ func (r *Reconciler) delete() error {
 		return fmt.Errorf("powering off vm is in progress, requeuing")
 	}
 
+	// At this point node should be evicted and vm powered off already.
+	// We need to check attached disks and ensure that all disks potentially related to PVs were detached
+	// to prevent possible data loss.
+	// Destroying a VM with attached disks might lead to data loss in case pvs are handled by the intree storage driver.
+	klog.V(3).Infof("Checking attached disks before vm destroy")
+	disks, err := vm.getAttachedDisks()
+	if err != nil {
+		return fmt.Errorf("%v: can not obtain virtual disks attached to the vm: %w", r.machine.GetName(), err)
+	}
+	// Currently, MAPI does not provide any API knobs to configure additional volumes for a VM.
```

Comment:
Have you considered what would happen if we changed this? What is the API like? Is the API a slice which is capped at 1 length? Wondering if it would be better to compare to the list length if that's the API

Reply:
There is no such api at all.

https://github.com/openshift/api/blob/master/machine/v1beta1/types_vsphereprovider.go#L53 - the only disks related knob there.

Reply:
Excellent, in which case, please ignore this comment

Reply:
~Another thing came to my mind which needs to be discussed and explored: Is it possible to have more than one disk in a template that we are cloning?~

Found an answer while writing down this comment:
https://github.com/openshift/machine-api-operator/blob/master/pkg/controller/vsphere/reconciler.go#L821

Mapi will prevent clone if there is more than one disk.